pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -45,15 +49,75 @@ func New(store storage.Store, prov provider) *Service {
 
 // Handle didexchange msg
 func (s *Service) Handle(msg dispatcher.DIDCommMsg) error {
-	// TODO add Handle logic
+	thid, err := threadID(msg.Payload)
+	if err != nil {
+		return err
+	}
+	current, err := s.currentState(thid)
+	if err != nil {
+		return err
+	}
+	next, err := stateFromMsgType(msg.Type)
+	if err != nil {
+		return err
+	}
+	if !current.CanTransitionTo(next) {
+		return errors.Errorf("invalid state transition: %s -> %s", current.Name(), next.Name())
+	}
+	// TODO call pre-transition listeners
```

Comment:
What are the GitHub issue numbers for these TODOs?

Reply:
@troyronda This was created initially https://github.com/hyperledger/aries-framework-go/issues/140

Reply:
Can you mention issue number in TODO?